stdlib/private/SwiftPrivateLibcExtras/Subprocess.c

```diff
@@ -0,0 +1,50 @@
+//===--- Subprocess.cpp - Subprocess Stubs --------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#include <spawn.h>
+#include <sys/types.h>
+#if defined(__APPLE__)
+#include <crt_externs.h> // for _NSGetEnviron
+#endif
+
+int swift_posix_spawn_file_actions_init(
+    posix_spawn_file_actions_t *file_actions) {
+  return posix_spawn_file_actions_init(file_actions);
+}
+
+int swift_posix_spawn_file_actions_destroy(
+    posix_spawn_file_actions_t *file_actions) {
+  return posix_spawn_file_actions_destroy(file_actions);
+}
+
+int swift_posix_spawn_file_actions_addclose(
+    posix_spawn_file_actions_t *file_actions, int filedes) {
+  return posix_spawn_file_actions_addclose(file_actions, filedes);
+}
+
+int swift_posix_spawn_file_actions_adddup2(
+    posix_spawn_file_actions_t *file_actions, int filedes, int newfiledes) {
+  return posix_spawn_file_actions_adddup2(file_actions, filedes, newfiledes);
+}
+
+int swift_posix_spawn(pid_t *__restrict pid, const char * __restrict path,
+                      const posix_spawn_file_actions_t *file_actions,
+                      const posix_spawnattr_t *__restrict attrp,
+                      char *const argv[__restrict],
+                      char *const envp[__restrict]) {
+  return posix_spawn(pid, path, file_actions, attrp, argv, envp);
+}
+
+#if defined(__APPLE__)
+char ***swift__NSGetEnviron(void) { return _NSGetEnviron(); }
```

Comment:
@gparker42 No, it is a part of the C++ spec.

N4431, [lex.name] p3:
>  Each identifier that contains a double underscore `__` or begins with an underscore followed by an uppercase letter is reserved to the implementation for any use.

@compnerd `swift_SwiftPrivateLibcExtras_NSGetEnviron`, I think.  It won't be confusing to drop one underscore, but this way we won't step into the reserved namespace.